crates/pop-cli/src/commands/up/contract.rs

```diff
@@ -438,15 +470,109 @@ mod tests {
 		Ok(())
 	}
 
+	#[tokio::test]
 	async fn get_upload_call_data_works() -> anyhow::Result<()> {
-		todo!()
+		let (contracts_node_process, port, temp_dir) = start_test_environment().await?;
+		let localhost_url = format!("ws://127.0.0.1:{}", port);
+		sleep(Duration::from_secs(20)).await;
+
+		let up_contract_opts = UpContractCommand {
+			path: Some(temp_dir.path().join("testing")),
+			constructor: "new".to_string(),
+			args: vec![],
+			value: "0".to_string(),
+			gas_limit: None,
+			proof_size: None,
+			salt: None,
+			url: Url::parse(&localhost_url).expect("given url is valid"),
+			suri: "//Alice".to_string(),
+			dry_run: false,
+			upload_only: true,
+			skip_confirm: true,
+			use_wallet: true,
+		};
+
+		let rpc_client = subxt::backend::rpc::RpcClient::from_url(&up_contract_opts.url).await?;
+		let client = subxt::OnlineClient::<SubstrateConfig>::from_rpc_client(rpc_client).await?;
+
+		// Retrieve call data based on the above command options.
+		let (retrieved_call_data, _) = match up_contract_opts.get_contract_data().await {
+			Ok(data) => data,
+			Err(e) => {
+				error(format!("An error occurred getting the call data: {e}"))?;
+				return Err(e);
+			},
+		};
+		// We have retrieved some payload.
+		assert!(!retrieved_call_data.is_empty());
+
+		// Craft encoded call data for an upload code call.
+		let contract_code = get_contract_code(up_contract_opts.path.as_ref()).await?;
+		let storage_deposit_limit: Option<u128> = None;
+		let upload_code = contract_extrinsics::extrinsic_calls::UploadCode::new(
+			contract_code,
+			storage_deposit_limit,
+			contract_extrinsics::upload::Determinism::Enforced,
+		);
+		let expected_call_data = upload_code.build();
+		let mut encoded_expected_call_data = Vec::<u8>::new();
+		expected_call_data
+			.encode_call_data_to(&client.metadata(), &mut encoded_expected_call_data)?;
+
+		// Retrieved call data and calculated match.
+		assert_eq!(retrieved_call_data, encoded_expected_call_data);
+
+		// Stop running contracts-node
+		stop_test_environment(&contracts_node_process.id().to_string())?;
+		Ok(())
 	}
 
+	#[tokio::test]
 	async fn get_instantiate_call_data_works() -> anyhow::Result<()> {
-		todo!()
-	}
+		let (contracts_node_process, port, temp_dir) = start_test_environment().await?;
+		let localhost_url = format!("ws://127.0.0.1:{}", port);
+		sleep(Duration::from_secs(20)).await;
+
+		let up_contract_opts = UpContractCommand {
+			path: Some(temp_dir.path().join("testing")),
+			constructor: "new".to_string(),
+			args: vec!["false".to_string()],
+			value: "0".to_string(),
+			gas_limit: None,
+			proof_size: None,
```

Comment:
would be nice if these were some value to ensure pop-cli isn't just taking the default None.

Reply:
Addressed in [86ab4b8](https://github.com/r0gue-io/pop-cli/pull/374/commits/86ab4b88eadfd6955982e12ca849fe0bcd2bab55)